File: data/multilang_comparison/program_julia/program_julia/src/func41.jl
include("program_julia_head.jl")
function func41(vars::JArray_param, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array20 = nothing
   if pCounter > 1
       array20 = vars.data[pCounter]
       pCounter -= 1
       array20.refC += 1
   else
       array20 = JArray(zeros(Int, 814), 814, 1, 20)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array20)
   params0.size += 1
   array21 = func46(params0, get_path(), loopsFactor);
   loopLimit8 = (50)/5 + 1
   for i in 1:loopLimit8 
      for i in 0:(array20.size-1) 
         array20.data[i+1] += 1
      end
   end
   for i in 1:array21.size
       if array21.data[i] == 50
               return array21
      end
   end
   array20.refC -= 1
   if array20.refC == 0
       empty!(array20.data)
       array20 = nothing
   end
   return array21;
end

File: data/multilang_comparison/program_go/program_go/src/func41.go
package main
import "math/rand"
var _ = rand.Intn
func Func41(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array20 *Array
   if pCounter > 0 {
       array20 = vars.Data[pCounter-1]
       pCounter--
       array20.RefC++
   } else {
       array20 = &Array{
           Size: 814,
           RefC: 1,
           Id: 20,
           Data: make([]uint32, 814),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array20
   var array21 *Array = Func46(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   var loop8 int = 0;
   var loopLimit8 int = (50)/5 + 1;
   for ; loop8 < loopLimit8; loop8++ {
      for i := 0; i < array20.Size; i++ {
          array20.Data[i]++
      }
   }
   for i := 0; i < array21.Size; i++ {
       if array21.Data[i] == 50 {
           return array21
       }
   }
   array20.RefC--
   if array20.RefC == 0 {
       array20.Data = nil
       array20 = nil
   }
   return array21;
}

